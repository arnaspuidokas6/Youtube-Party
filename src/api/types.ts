export interface FetchVideoParameters {
    limit?: number;
    searchValue: string;
}

export interface IVideo {
    snippet: {
        title: string;
        channelTitle: string;
        thumbnails: { medium: { url: string } };
    };
}
