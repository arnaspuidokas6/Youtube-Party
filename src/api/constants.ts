export const YOUTUBE_API_KEY = 'AIzaSyCp5qn34dm4AOsHcFIjTcJ1YgKM7UZlkSM';
export const BASE_YOUTUBE_URL = 'https://youtube.googleapis.com/youtube/v3/search';
export const DEFAULT_LIST_ITEMS = 20;
