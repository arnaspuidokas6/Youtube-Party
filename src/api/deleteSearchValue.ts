import axios from 'axios';

export const deleteSearchValue = async (searchValueId: string) => {
    const res = await axios.delete(`http://localhost:8080/api/search/${searchValueId}`);
    return res;
};
