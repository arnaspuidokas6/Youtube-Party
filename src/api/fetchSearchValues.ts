import axios from 'axios';

export const fetchSearchValues = async () => {
    const res = await axios.get(`http://localhost:8080/api/search`);
    console.log(res);
    return res;
};
