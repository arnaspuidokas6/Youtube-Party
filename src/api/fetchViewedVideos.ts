import axios from 'axios';
import { ViewedVideo } from './postViewedVideo';

export const fetchViewedVideos = async (): Promise<ViewedVideo[]> => {
    return await axios.get(`http://localhost:8080/api/video`);
};
