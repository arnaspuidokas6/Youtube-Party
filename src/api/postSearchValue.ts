import axios from 'axios';

interface IPostSearch {
    searchValue: string;
}

export const postSearchValues = async ({ searchValue }: IPostSearch) =>
    await axios.post(`http://localhost:8080/api/search`, { searchValue });
