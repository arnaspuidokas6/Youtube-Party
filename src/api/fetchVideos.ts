import axios from 'axios';
import { YOUTUBE_API_KEY, BASE_YOUTUBE_URL } from './constants';
import { FetchVideoParameters, IVideo } from './types';

export const fetchVideos = async ({ limit = 20, searchValue }: FetchVideoParameters): Promise<IVideo[]> => {
    const res = await axios.get(
        `${BASE_YOUTUBE_URL}?part=snippet&maxResults=${`${limit}`}&q=${searchValue}&key=${YOUTUBE_API_KEY}`,
    );

    return res?.data?.items;
};
