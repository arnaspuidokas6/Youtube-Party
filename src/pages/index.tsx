import React, { createContext, Dispatch, FC, SetStateAction, useContext } from 'react';
import { IVideo } from '../api/types';

export const EMPTY_VIDEOS_LIST = [];
export const DEFAULT_SELECTED_ITEM = {} as IVideo;
const EMPTY_VOID = () => {};

interface IVideosContext {
    setSearchValue: Dispatch<SetStateAction<string>>;
    searchValue?: string;
    setSelectedVideo: Dispatch<SetStateAction<IVideo>>;
    videosList: IVideo[];
    selectedVideo: IVideo;
}

export const VideosListContext = createContext<IVideosContext>({
    setSearchValue: EMPTY_VOID,
    setSelectedVideo: EMPTY_VOID,
    videosList: EMPTY_VIDEOS_LIST,
    selectedVideo: DEFAULT_SELECTED_ITEM,
});

export const useVideosContext = () => {
    const context = useContext(VideosListContext);
    if (!context) {
        throw new Error("This component can't be used outside VideosList component");
    }
    return context;
};

export const VideosListPage: FC<IVideosContext> = ({
    setSearchValue,
    setSelectedVideo,
    videosList,
    selectedVideo,
    children,
    searchValue,
}) => {
    const context: IVideosContext = {
        setSearchValue,
        videosList,
        setSelectedVideo,
        selectedVideo,
        searchValue,
    };
    return <VideosListContext.Provider value={context}>{children}</VideosListContext.Provider>;
};
