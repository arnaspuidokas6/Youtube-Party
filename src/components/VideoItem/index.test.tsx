import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { VideoItem } from '.';
import { IVideo } from '../../api/types';
import { VideosListContext } from '../../pages';

describe('Video Item', () => {
    const mockVideo: IVideo = {
        snippet: {
            title: 'Random video title',
            channelTitle: 'Random channel title',
            thumbnails: { medium: { url: 'https://www.youtube.com/watch?v=EFVcKWFajHI' } },
            description: 'Random description',
        },
        id: { videoId: 'string' },
    };
    const videosList: IVideo[] = [];
    const selectedVideo = {} as IVideo;
    const setSearchValue = jest.fn();
    const setSelectedVideo = jest.fn().mockResolvedValueOnce(true);
    it('should render video', () => {
        const { baseElement } = render(
            <VideosListContext.Provider value={{ setSearchValue, setSelectedVideo, videosList, selectedVideo }}>
                <VideoItem video={mockVideo} />
            </VideosListContext.Provider>,
        );
        expect(baseElement).toMatchSnapshot();
    });

    it('should allow to select when clicked', () => {
        render(
            <VideosListContext.Provider value={{ setSearchValue, setSelectedVideo, videosList, selectedVideo }}>
                <VideoItem video={mockVideo} />
            </VideosListContext.Provider>,
        );
        userEvent.click(screen.getByTestId('video-item'));
        expect(setSelectedVideo).toHaveBeenCalledTimes(1);
        expect(setSelectedVideo).toHaveBeenCalledWith(mockVideo);
    });
});
