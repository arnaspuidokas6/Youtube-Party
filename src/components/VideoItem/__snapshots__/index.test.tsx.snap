// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Item should render video 1`] = `
<body>
  <div>
    <div
      class="flex flex-wrap w-full mb-3"
    >
      <img
        class="w-1/2 cursor-pointer"
        src="https://www.youtube.com/watch?v=EFVcKWFajHI"
      />
      <div
        class="w-1/2 pl-2"
      >
        <h3
          class="text-base mb-2"
        >
          Random video title
        </h3>
        <p
          class="text-sm text-grey-darken mb-1"
        >
          Random channel title
        </p>
      </div>
    </div>
  </div>
</body>
`;
